slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/dao/SignedAttestationsDao.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.web3signer.slashingprotection.dao;
+
+import java.util.Optional;
+
+import org.apache.tuweni.units.bigints.UInt64;
+import org.jdbi.v3.core.Handle;
+
+public class SignedAttestationsDao {
+
+  public Optional<SignedAttestation> findExistingAttestation(
+      final Handle handle, final long validatorId, final UInt64 targetEpoch) {
+    return handle
+        .createQuery(
+            "SELECT validator_id, source_epoch, target_epoch, signing_root "
+                + "FROM signed_attestations WHERE validator_id = ? AND target_epoch = ?")
+        .bind(0, validatorId)
+        .bind(1, targetEpoch)
+        .mapToBean(SignedAttestation.class)
+        .findFirst();
+  }
+
+  public Optional<SignedAttestation> findSurroundingAttestation(
+      final Handle handle,
+      final long validatorId,
+      final UInt64 sourceEpoch,
+      final UInt64 targetEpoch) {
+    return handle
+        .createQuery(
+            "SELECT validator_id, source_epoch, target_epoch, signing_root "
+                + "FROM signed_attestations "
+                + "WHERE validator_id = ? AND source_epoch < ? AND target_epoch > ? "
```

Comment:
Should they be simply "<" and ">" or is it "<=" and ">="?
I.e. not all possible cases are covered with these queries (I don't think)

Reply:
No I don't think it should be. From https://github.com/ethereum/eth2.0-specs/blob/dev/specs/phase0/beacon-chain.md#is_slashable_attestation_data I think this is correct.